### PR TITLE
Prepare to serve static assets via Cloudfront

### PR DESCRIPTION
**Pull request by gregdingle**
# Description

This makes idseq rails app use a new env var `CLOUDFRONT_ENDPOINT` if available on init .

The future values should be `assets.staging.idseq.net` and `assets.idseq.net`. 

# Notes

I followed the guides https://aws.amazon.com/blogs/developer/caching-the-rails-asset-pipeline-with-amazon-cloudfront/ and https://aws.amazon.com/premiumsupport/knowledge-center/cloudfront-distribution-serve-content/

# Tests

* start locally without `CLOUDFRONT_ENDPOINT`, see assets served locally
* set `CLOUDFRONT_ENDPOINT` to `assets.staging.idseq.net`, see assets served remotely, see site load correctly

### CODE REVIEW

File: app/assets/src/components/views/samples/ProjectSettingsModal/PublicProjectConfirmationModal.jsx [hunk 1]
```diff
@@ -52,9 +52,7 @@ class PublicProjectConfirmationModal extends React.Component {
                 or{" "}
                 <span
                   className={cs.link}
-                  onClick={() =>
-                    window.open("https://assets.idseq.net/TermsOfService.pdf")
-                  }
+                  onClick={() => window.open("https://idseq.net/terms")}
```
Comment (gregdingle) [inline app/assets/src/components/views/samples/ProjectSettingsModal/PublicProjectConfirmationModal.jsx]: I found this old (broken) reference while searching the code base

File: config/environments/staging.rb [hunk 2]
```diff
@@ -82,7 +79,9 @@
     domain: 'mg.idseq.net',
   }
 
-  config.action_controller.asset_host = 'staging.idseq.net'
+  # Enable serving of images, stylesheets, and JavaScripts from an asset server.
+  # We configure IDseq to use cloudfront CDN when available.
+  config.action_controller.asset_host = ENV['CLOUDFRONT_ENDPOINT'] || 'staging.idseq.net'
```
Comment (MarkAZhang) [inline config/environments/staging.rb]: Nit: Thoughts on naming this `STATIC_ASSETS_ENDPOINT` instead? Just to keep it more generic.

Comment (gregdingle) [reply config/environments/staging.rb]: I hadn't considered that but I think value of being consistent with AWS docs outweighs the value of being generic. 

FWIW, we've also been consistently using AWS terms in our env vars, such as `ENV['REDISCLOUD_URL']`.

**Comment by gregdingle** [comment]
I'm going to wait on the terraform PR to merge this.

**Comment by jshoe** [comment]
What assets are we going to be able to get onto CloudFront? All the Javascript and CSS?

**Comment by gregdingle** [comment]
> What assets are we going to be able to get onto CloudFront? All the Javascript and CSS?

Yes, everything served from `/assets` path.

**Comment by gregdingle** [comment]
I decided to remove the old useless assets.idseq.net s3 bucket in https://github.com/chanzuckerberg/idseq-infra/pull/269 . So I moved the only two files in the bucket into the web repo, so they can still be referenced by the same URLs. For example, https://assets.staging.idseq.net/logo-prod.png will go to https://staging.idseq.net/assets/Logo_Black.png 

I tested it with https://staging.idseq.net/assets/logo-prod.png.

**Pull request merged**